sbi/inference/potentials/likelihood_based_potential.py

```diff
@@ -119,19 +120,28 @@ def _log_likelihoods_over_trials(
         log_likelihood_trial_sum: log likelihood for each parameter, summed over all
             batch entries (iid trials) in `x`.
     """
-    # unsqueeze to ensure that the x-batch dimension is the first dimension for the
-    # broadcasting of the density estimator.
-    x = torch.as_tensor(x).reshape(-1, x.shape[-1]).unsqueeze(1)
+    # Shape of `x` is (iid_dim, *event_shape).
+    x = reshape_to_iid_batch_event(x, event_shape=x.shape[1:], leading_is_iid=True)
+
+    # Match the number of `x` to the number of conditions (`theta`). This is important
+    # if the potential is simulataneously evaluated at multiple `theta` (e.g.
+    # multi-chain MCMC).
+    theta_batch_size = theta.shape[0]
+    x = x.expand(-1, theta_batch_size, -1)
```

Comment:
Before x is of shape `[iid_dim, *event_shape]`, but this will flatten the `event_shape` into a single dimension, which could potentially break 2d images, no?

Reply:
This is a great catch, thanks a ton!